scrape/scrape.go

```diff
@@ -1273,8 +1280,30 @@ loop:
 		if ok {
 			err = app.AddFast(ce.ref, t, v)
 			_, err = sl.checkAddError(ce, met, tp, err, &sampleLimitErr, &appErrs)
+			switch err {
+			case nil:
+				if tp == nil {
+					sl.cache.trackStaleness(ce.hash, ce.lset)
+				}
+				// todo: This smells funny.
+				if hasExemplar := p.Exemplar(&e); hasExemplar {
+					if err := exemplarApp.AddExemplar(ce.lset, t, e); err != nil {
```

Comment:
Should we have sanity checks on the exemplar timestamp, like we do for scraping generally?

Reply:
Yes I will look at how we're doing that. My assumption was that the majority of the time an exemplar would have the same timestamp as a scraped sample so I didn't put time into doing validation of the timestamp here yet.

Reply:
It'd be very rare that it's have the same timestamp, usually it'll be in the past (if there is a timestamp) and it may even appear a bit in the future.

Reply:
I think this can be marked as resolved now with the checks in `exemplar.go` `addExemplar` and the `ErrOutOfOrderExemplar`. Let me know if you disagree.

Reply:
No, it's not right. The check should be that it's not more than 10m in the future.

Reply:
I'm not seeing where we enforce that same `not more than 10m in the future` for regular samples. I just see `minValidTime` in `head.go`.

Reply:
https://github.com/prometheus/prometheus/blob/275f7e7766f80648d6e63ed968685f3963b494e9/scrape/scrape.go#L223